### PR TITLE
Add some missing field readers to XFontStruct

### PR DESCRIPTION
**Pull request by kanaihiroki**
Hello, this PR adds the reader functions which seems to be missing.

### CODE REVIEW

File: Graphics/X11/Xlib/Font.hsc [hunk 1]
```diff
@@ -35,7 +37,7 @@ module Graphics.X11.Xlib.Font(
 import Graphics.X11.Types
 import Graphics.X11.Xlib.Types
 
-import Foreign (Ptr, Int32, alloca, allocaBytes, peekByteOff, Word16, #{type unsigned long}, peek, throwIfNull)
+import Foreign (Ptr, Int32, alloca, allocaBytes, peekByteOff, Word16, #{type unsigned long}, peek, plusPtr, throwIfNull)
```
Comment (geekosaur) [inline Graphics/X11/Xlib/Font.hsc:40]: I don't see `plusPtr` being used?

**Comment by geekosaur** [comment]
Looks like the failures are because of hvr's PPA. On the one hand, we could fix the references; on the other, it's well past time to stop relying on them and switch to ghcup.

**Comment by slotThe** [comment]
I think we just use what `haskell-ci` uses as its default value, though there seem to be plans to update that (see https://github.com/haskell-CI/haskell-ci/pull/685)

**Comment by liskin** [comment]
> I think we just use what `haskell-ci` uses as its default value, though there seem to be plans to update that (see [haskell-CI/haskell-ci#685](https://github.com/haskell-CI/haskell-ci/pull/685))

When that's merged we'll still get failures with 7.10 I believe, which we still support because of https://github.com/xmonad/X11/issues/80. It's been 2 years since that issue was filed so I guess we can drop support for GHC < 8, we just need to remember to bump the `base` bounds so that cabal picks the previous version when someone tries to use an old GHC.

File: Graphics/X11/Xlib/Font.hsc (hunk 1 above)
Comment (kanaihiroki) [reply Graphics/X11/Xlib/Font.hsc:40]: It seems that `plusPtr` is used at expanded code of `ptr` directive.
When `plusPtr` is dropped, following compile error has occurred. 

```
Graphics/X11/Xlib/Font.hsc:121:47: error:
    • Variable not in scope:
        plusPtr :: Ptr FontStruct -> t1 -> Ptr CharStruct
    • Perhaps you want to add ‘plusPtr’ to the import list
      in the import of ‘Foreign’ (Graphics/X11/Xlib/Font.hsc:40:1-96).
    |
121 |         peekCharStruct $ #{ptr XFontStruct,min_bounds} fs
    |    
```

**Review by geekosaur — APPROVED** [review]

**Pull request merged**